src/sentry/discover/endpoints/discover_key_transactions.py

```diff
@@ -0,0 +1,119 @@
+from __future__ import absolute_import
+
+from django.db import transaction
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.bases import OrganizationEventsV2EndpointBase
+from sentry.api.bases.organization import OrganizationPermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.discover.models import KeyTransaction
+from sentry.discover.endpoints.serializers import KeyTransactionSerializer
+from sentry.snuba.discover import query
+
+
+class KeyTransactionEndpoint(OrganizationEventsV2EndpointBase):
+    permission_classes = (OrganizationPermission,)
+
+    def has_feature(self, request, organization):
+        return features.has("organizations:performance-view", organization, actor=request.user)
+
+    def post(self, request, organization):
+        """ Create A Key Transaction """
+        if not self.has_feature(request, organization):
+            return Response(status=404)
+
+        projects = self.get_projects(request, organization)
+
+        if len(projects) != 1:
+            return Response({"detail": "Only 1 project per Key Transaction"}, status=400)
+        project = projects[0]
+
+        base_filter = {"organization": organization, "project": project, "owner": request.user}
+
+        with transaction.atomic():
+            serializer = KeyTransactionSerializer(data=request.data, context=base_filter)
+            if serializer.is_valid():
+                data = serializer.validated_data
+                base_filter["transaction"] = data["transaction"]
+
+                KeyTransaction.objects.create(**base_filter)
+                return Response(status=201)
+            else:
+                return Response(serializer.errors, status=400)
+
+    def get(self, request, organization):
+        """ Get the paged Key Transactions for a user """
+        if not self.has_feature(request, organization):
+            return self.response(status=404)
+
+        params = self.get_filter_params(request, organization)
+        fields = request.GET.getlist("field")[:]
+        orderby = self.get_orderby(request)
+
+        queryset = KeyTransaction.objects.filter(organization=organization, owner=request.user)
```

Comment:
How do users share key transactions?

Reply:
So this wasn't intended to be shareable yet. But It shouldn't be too hard to extend this get function to allow sharing, or to get key transactions for an entire org.